gson-flatten/src/main/java/org/itishka/gsonflatten/FlattenTypeAdapterFactory.java

```diff
@@ -125,13 +186,13 @@ public T read(JsonReader in) throws IOException {
         return cache;
     }
 
-    private static class FlattenCacheItem {
+    protected static class FlattenCacheItem {
```

Comment:
private -> protected on these because you're accessing them from an inner class, so we're saving the compiler synthesizing a method. Entirely unrelated to the array stuff.